client/llb/state.go

```diff
@@ -274,12 +275,42 @@ func (s State) Run(ro ...RunOption) ExecState {
 }
 
 func (s State) File(a *FileAction, opts ...ConstraintsOpt) State {
+	res, _ := s.FileOptimize(a, false, opts...)
+	return res
+}
+
+// FileOptimize supports optimization.
+// The optimized result may not contain FileOp.
+//
+// e.g., llb.Scratch().File(llb.Copy(llb.Git(), "/", "/")) --> llb.Git()
+func (s State) FileOptimize(a *FileAction, attemptOptimization bool, opts ...ConstraintsOpt) (State, bool) {
```

Comment:
Why another public function?

Reply:
Because it might be safer not to let `File()` optimize the LLB ops automatically.

Reply:
What possible problems do you see with enabling this by default?

I think we should either enable it by default or do this detection in Dockerfile frontend instead.

Reply:
A frontend developer may get confused when `File()` actually does not create any FileOp by default